overlord/configstate/configcore/prompting.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build !nomanagers
+// +build !nomanagers
+
+/*
+ * Copyright (C) 2017-2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/snapcore/snapd/features"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+)
+
+func doExperimentalApparmorPromptProfileRegeneration(c RunTransaction, opts *fsOnlyContext) error {
+	st := c.State()
+
+	var prompting bool
+	err := c.Get("core", "experimental."+features.AppArmorPrompting.String(), &prompting)
+	if err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	var prevPrompting bool
+	err = c.GetPristine("core", "experimental."+features.AppArmorPrompting.String(), &prevPrompting)
```

Comment:
`ConfigOption` gives us what the name we need, if you prefer that
```suggestion
    _, confName := features.AppArmorPrompting.ConfigOption()

	var prompting bool
	err := c.Get("core", confName, &prompting)
	if err != nil && !config.IsNoOption(err) {
		return err
	}
	var prevPrompting bool
	err = c.GetPristine("core", confName, &prevPrompting)
```